server/db-middleware/check-grade-api.js

```diff
@@ -32,28 +39,142 @@ const isGraidingPossibleIndividual = function (assignment) {
 
 const isGraidingPossibleGroup = function (assignment) {
   let stdAmount;
-  return generalApi.getStudentsByGroup(assignment.groupId)
+  let requestAmount;
+  return TestAssignment
+    .findById(assignment._id)
+    .then(assign => generalApi.getStudentsByGroup(assign.groupId))
     .then((std) => {
       stdAmount = std.length;
+      return CheckRequest
+        .countDocuments({ assignmentId: assignment._id, status: REQUEST_STATUS_PENDING });
+    })
+    .then((check) => {
+      requestAmount = !check;
     })
     .then(() => TestSubmission
       .countDocuments({ assignmentId: assignment._id, status: REQUEST_STATUS_CHECKED }))
     .then((submissionAmount) => {
-      if (stdAmount === submissionAmount ||
-        (assignment.deadline !== null && assignment.deadline < new Date().getTime())
+      if ((stdAmount === submissionAmount && requestAmount) ||
+        (assignment.deadline !== undefined && assignment.deadline < new Date().getTime())
       ) return true;
       return false;
     });
 };
 
-apiModule.isGraidingPossible = function (assignmentId) {
+const updateQuestion = function (questId) {
+  return Question
+    .findById(questId)
+    .select('peopleTested peopleAnswered difficulty')
+    .then((info) => {
+      let diff;
+      if (+info.peopleTested !== 0) {
```

Comment:
No need in unary operator, it is specified in schema as number, so no casting required